tests/components/here_weather/test_config_flow.py

```diff
@@ -0,0 +1,172 @@
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}
+        )
+        assert result["type"] == "form"
+        config = {
+            CONF_API_KEY: "test",
+            CONF_NAME: DOMAIN,
+            CONF_LATITUDE: "40.79962",
+            CONF_LONGITUDE: "-73.970314",
+        }
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], config
+        )
+        assert result["type"] == "create_entry"
+
+        await hass.async_block_till_done()
+        state = hass.states.get("weather.here_weather_forecast_7days_simple")
+        assert state
+
+
+async def test_unauthorized(hass):
+    """Test handling of an unauthorized api key."""
+    with patch(
+        "herepy.DestinationWeatherApi.weather_for_coordinates",
+        side_effect=herepy.UnauthorizedError("Unauthorized"),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}
+        )
+        assert result["type"] == "form"
+        config = {
+            CONF_API_KEY: "test",
+            CONF_NAME: DOMAIN,
+            CONF_LATITUDE: "40.79962",
+            CONF_LONGITUDE: "-73.970314",
+        }
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], config
+        )
+        assert result["type"] == "form"
+        assert result["errors"]["base"] == "unauthorized"
+
+
+async def test_invalid_request(hass):
+    """Test handling of an invalid request."""
+    with patch(
+        "herepy.DestinationWeatherApi.weather_for_coordinates",
+        side_effect=herepy.InvalidRequestError("Invalid"),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}
+        )
+        assert result["type"] == "form"
+        config = {
+            CONF_API_KEY: "test",
+            CONF_NAME: DOMAIN,
+            CONF_LATITUDE: "40.79962",
+            CONF_LONGITUDE: "-73.970314",
+        }
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], config
+        )
+        assert result["type"] == "form"
+        assert result["errors"]["base"] == "invalid_request"
+
+
+async def test_options(hass):
+    """Test the options flow."""
+    with patch(
+        "herepy.DestinationWeatherApi.weather_for_coordinates",
+        side_effect=mock_weather_for_coordinates,
+    ):
+        entry = MockConfigEntry(
+            domain=DOMAIN,
+            data={
+                CONF_API_KEY: "test",
+                CONF_NAME: DOMAIN,
+                CONF_LATITUDE: "40.79962",
+                CONF_LONGITUDE: "-73.970314",
+            },
+        )
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
```

Comment:
Why do we need to start home assistant?

Reply:
Removed the line in https://github.com/home-assistant/core/pull/28910/commits/bf7f9adf72fba862ce9335013e48e3bef1b74b1e